src/FuseGPUThreadLoops.cpp

```diff
@@ -868,7 +1087,11 @@ class FuseGPUThreadLoops : public IRMutator {
                      << loop << "\n\n";
 
             // Mutate the inside of the kernel
-            return FuseGPUThreadLoopsSingleKernel(block_size, shared_mem).mutate(loop);
+            loop = FuseGPUThreadLoopsSingleKernel(block_size, shared_mem).mutate(loop);
```

Comment:
nit: I found the variable name "shared_mem" slightly confusing since it can now be shared memory *or* global memory. Maybe call it `shared_alloc`.